lib/server.js

```diff
@@ -15,385 +15,371 @@ const userAgent = require('./userAgent')
-
-  const metricsEvents = require('./metrics/events')(log, config)
-  server.decorate('request', 'emitMetricsEvent', metricsEvents.emit)
-  server.decorate('request', 'emitRouteFlowEvent', metricsEvents.emitRouteFlowEvent)
-
-  server.stat = function() {
-    return {
-      stat: 'mem',
-      rss: server.load.rss,
-      heapUsed: server.load.heapUsed
+
+         await server.register({
+                plugin: require('hapi-hpkp'),
+                options: hpkpOptions
+         })
+
     }
-  }
 
-  return server
+//register auth-hawk
+    await server.register(require('hapi-auth-hawk'))
+
+    server.auth.strategy(
+        'sessionToken',
+        'hawk',
+        {
+            getCredentialsFunc: makeCredentialFn(db.sessionToken.bind(db)),
+            hawk: hawkOptions
+        }
+    )
+    server.auth.strategy(
+        'keyFetchToken',
+        'hawk',
+        {
+            getCredentialsFunc: makeCredentialFn(db.keyFetchToken.bind(db)),
+            hawk: hawkOptions
+        }
+    )
+    server.auth.strategy(
+        // This strategy fetches the keyFetchToken with its
+        // verification state. It doesn't check that state.
+        'keyFetchTokenWithVerificationStatus',
+        'hawk',
+        {
+            getCredentialsFunc: makeCredentialFn(db.keyFetchTokenWithVerificationStatus.bind(db)),
+            hawk: hawkOptions
+        }
+    )
+    server.auth.strategy(
+        'accountResetToken',
+        'hawk',
+        {
+            getCredentialsFunc: makeCredentialFn(db.accountResetToken.bind(db)),
+            hawk: hawkOptions
+        }
+    )
+    server.auth.strategy(
+        'passwordForgotToken',
+        'hawk',
+        {
+            getCredentialsFunc: makeCredentialFn(db.passwordForgotToken.bind(db)),
+            hawk: hawkOptions
+        }
+    )
+    server.auth.strategy(
+        'passwordChangeToken',
+        'hawk',
+        {
+            getCredentialsFunc: makeCredentialFn(db.passwordChangeToken.bind(db)),
+            hawk: hawkOptions
+        }
+    )
+    await server.register(require('hapi-fxa-oauth'))
+
+    server.auth.strategy('oauthToken', 'fxa-oauth', config.oauth)
+    // routes should be registered after all auth strategies have initialized:
+    // ref: http://hapijs.com/tutorials/auth
+
+    server.route(routes)
+    return server
 }
 
 function defineLazyGetter (object, key, getter) {
-  let value
-  Object.defineProperty(object, key, {
-    get () {
-      if (! value) {
-        value = getter()
-      }
-
-      return value
-    }
-  })
+    let value
+    Object.defineProperty(object, key, {
+        get () {
+            if (! value) {
+                value = getter()
+            }
+
+            return value
+        }
+    })
 }
 
 module.exports = {
-  create: create,
-  // Functions below exported for testing
-  _trimLocale: trimLocale,
-  _logEndpointErrors: logEndpointErrors
+    create: create,
```

Comment:
not sure why this changed?  I Think this file needs to be fixed back to 2 space indents instead of 4